packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env node
+
+/**
+ * automatically creates a module for any CloudFormation namespaces that do not
+ * have an AWS construct library.
+ */
+
+import child_process = require('child_process');
+import fs = require('fs');
+import path = require('path');
+import { promisify } from 'util';
+import cfnspec = require('../lib');
+
+const exist = promisify(fs.exists);
+const mkdir = promisify(fs.mkdir);
+const writeFile = promisify(fs.writeFile);
+const readdir = promisify(fs.readdir);
+const copyFile = promisify(fs.copyFile);
+
+// don't be a prude:
+// tslint:disable:no-console
+// tslint:disable:object-literal-key-quotes
+
+async function main() {
+  const root = path.join(__dirname, '..', '..');
+  if (path.basename(root) !== '@aws-cdk') {
+    throw new Error(`Something went wrong. We expected ${root} to be the "packages/@aws-cdk" directory. Did you move me?`);
+  }
+
+  const version = require('../package.json').version;
+
+  // iterate over all cloudformation namespaces
+  for (const namespace of cfnspec.namespaces()) {
+    const [ moduleFamily, moduleBaseName ] = namespace.split('::');
+
+    const moduleName = `${moduleFamily}-${moduleBaseName}`.toLocaleLowerCase();
+    const pacakgePath = path.join(root, moduleName);
+
+    // we already have a module for this namesapce, move on.
+    if (await exist(pacakgePath)) {
+      continue;
+    }
+
+    const lowcaseModuleName = moduleBaseName.toLocaleLowerCase();
+    const packageName = `@aws-cdk/${moduleName}`;
+
+    // dotnet names
+    const dotnetPackage = `Amazon.CDK.${moduleFamily}.${moduleBaseName}`;
+
+    // java names
+    const javaGroupId = 'software.amazon.awscdk';
+    const javaPackage = moduleFamily === 'AWS'
+      ? `services.${lowcaseModuleName}`
```

Comment:
I like multi-line ternary assignment r-values to have the `?` and `:` column-aligned on the assignment `=`. But maybe that's just me? :)

Reply:
I guess it's a matter of style. I kinda like it like this if that's okay :-)